apis/network/v1alpha3/types.go

```diff
@@ -218,10 +218,198 @@ type Subnet struct {
 }
 
 // +kubebuilder:object:root=true
-
 // SubnetList contains a list of Subnet items
 type SubnetList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []Subnet `json:"items"`
 }
+
+//Network Security Group structs
+// SecurityRuleProtocol enumerates the values for security rule protocol.
+type SecurityRuleProtocol string
+
+// ApplicationSecurityGroupPropertiesFormat application security group properties.
+type ApplicationSecurityGroupPropertiesFormat struct {
+	// ResourceGUID - READ-ONLY; The resource GUID property of the application security group resource. It uniquely identifies a resource, even if the user changes its name or migrate the resource across subscriptions or resource groups.
+	ResourceGUID string `json:"resourceGuid,omitempty"`
+	// ProvisioningState - READ-ONLY; The provisioning state of the application security group resource. Possible values are: 'Succeeded', 'Updating', 'Deleting', and 'Failed'.
+	ProvisioningState string `json:"provisioningState,omitempty"`
+}
+
+// ApplicationSecurityGroup an application security group in a resource group.
+type ApplicationSecurityGroup struct {
+	// ApplicationSecurityGroupPropertiesFormat - Properties of the application security group.
+	Properties ApplicationSecurityGroupPropertiesFormat `json:"properties,omitempty"`
+	// Etag - READ-ONLY; A unique read-only string that changes whenever the resource is updated.
+	Etag string `json:"etag,omitempty"`
+	// ID - Resource ID.
+	ID string `json:"id,omitempty"`
+	// Name - READ-ONLY; Resource name.
+	Name string `json:"name,omitempty"`
+	// Type - READ-ONLY; Resource type.
+	Type string `json:"type,omitempty"`
+	// Location - Resource location.
+	Location string `json:"location,omitempty"`
+}
+
+// SecurityRuleAccess enumerates the values for security rule access.
+type SecurityRuleAccess string
+
+// SecurityRuleDirection enumerates the values for security rule direction.
+type SecurityRuleDirection string
+
+// SecurityRulePropertiesFormat security rule resource.
+type SecurityRulePropertiesFormat struct {
+	// Description - A description for this rule. Restricted to 140 chars.
+	Description string `json:"description,omitempty"`
```

Comment:
All optional spec fields should be pointer types.
```suggestion
	Description *string `json:"description,omitempty"`
```

Reply:
We have implemented this change